### PR TITLE
Add includeSystemOrg param to applications, subscriptions and application-key-mappings APIs

### PR DESCRIPTION
**Pull request by JanakaSandaruwan**
https://github.com/wso2-enterprise/choreo/issues/22160

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java [hunk 1]
```diff
@@ -114,6 +114,31 @@ public List<Subscription> getAllSubscriptionsByOrganization(String organization)
...
     }
 
+    /*
+     * This method can be used to retrieve all the APIs of a given organization in the database
+     *
+     * @param organization : organization Id
+     * @param systemOrg : organization Id of system org
+     * @return {@link List<Subscription>}
+     * */
+    public List<Subscription> getAllSubscriptionsByOrganization(String organization, String systemOrg) throws APIManagementException {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java:124]:
```suggestion
    public List<Subscription> getAllSubscriptionsOfDataplane(String dpOrg, String systemOrg) throws APIManagementException {
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java [hunk 2]
```diff
@@ -516,6 +541,29 @@ public List<Application> getAllApplications(String organization) {
         return applications;
     }
 
+    /*
+     * This method can be used to retrieve all the Applications of a given tenant in the database
+     * @param tenantId : tenant Id
+     * @param systemOrg: organization Id of system org
+     * @return {@link Subscription}
+     * */
+    public List<Application> getAllApplications(String organization, String systemOrg) {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java:550]:
```suggestion
    public List<Application> getAllApplicationsOfDataplane(String dpOrg, String systemOrg) {
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java [hunk 3]
```diff
@@ -582,6 +630,25 @@ public List<ApplicationKeyMapping> getAllApplicationKeyMappingsByOrganization(St
         return keyMappings;
     }
 
+    public List<ApplicationKeyMapping> getAllApplicationKeyMappingsByOrganization(String organization, String systemOrg) {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java:633]:
```suggestion
    public List<ApplicationKeyMapping> getAllApplicationKeyMappingsOfDataplane(String dpOrg, String systemOrg) {
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java [hunk 4]
```diff
@@ -516,6 +541,29 @@ public List<Application> getAllApplications(String organization) {
...
+        try (Connection conn = APIMgtDBUtil.getConnection();
+             PreparedStatement ps =
+                     conn.prepareStatement(SubscriptionValidationSQLConstants.GET_APPLICATIONS_BY_ORGANIZATION_SQL_WITH_SYSTEM_APPS)) {
+            ps.setString(1, organization);
+            ps.setString(2,systemOrg);
+            try (ResultSet resultSet = ps.executeQuery()) {
+                addToApplicationList(applications, resultSet);
+            }
+        } catch (SQLException e) {
+            log.error("Error in loading Applications for organization : " + organization + " including system org", e);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java:562]: Let's throw the errors. Lets fix the other places too

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SubscriptionValidationSQLConstants.java [hunk 5]
```diff
@@ -19,7 +19,7 @@
 package org.wso2.carbon.apimgt.impl.dao.constants;
 
 public class SubscriptionValidationSQLConstants {
-
+    public static final String IS_SYSTEM_APP_ATTRIBUTE = "isSystemApp";
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SubscriptionValidationSQLConstants.java:22]: Move to APIConstants

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java (hunk 1 above)
Comment (JanakaSandaruwan) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/12101/commits/202fb123e6ff5128e5b52ec84b71d792457a2119

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java (hunk 2 above)
Comment (JanakaSandaruwan) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/12101/commits/202fb123e6ff5128e5b52ec84b71d792457a2119

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java (hunk 3 above)
Comment (JanakaSandaruwan) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/12101/commits/202fb123e6ff5128e5b52ec84b71d792457a2119

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java (hunk 4 above)
Comment (JanakaSandaruwan) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/12101/commits/202fb123e6ff5128e5b52ec84b71d792457a2119

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SubscriptionValidationSQLConstants.java (hunk 5 above)
Comment (JanakaSandaruwan) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SubscriptionValidationSQLConstants.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/12101/commits/202fb123e6ff5128e5b52ec84b71d792457a2119

**Review by malinthaprasan — APPROVED** [review]

**Pull request merged**